src/mlir/generator.cc

```diff
@@ -564,14 +569,7 @@ namespace mlir::verona
   llvm::Expected<ReturnValue> Generator::parseContinue(const ::ast::Ast& ast)
   {
     assert(isContinue(ast) && "Bad node");
-    // Nested loops have multiple heads, we only care about the last one
-    if (!loopTable.inScope("head"))
-      return parsingError("Continue without a loop", getLocation(ast));
-    auto head = loopTable.lookup("head");
-    mlir::ValueRange empty{};
-    // We assume the continue is the last operation in its basic block
-    // and that was checked by the parser
-    builder.create<mlir::BranchOp>(getLocation(ast), head, empty);
+    builder.create<mlir::verona::ContinueOp>(getLocation(ast));
```

Comment:
This will allow, at the parser level, a `continue` outside of any loop, which will trip MLIR verifiers. I think we should check the presence of a loop here, such that we never generate syntactically invalid MLIR.

Same for break.

Reply:
The AST will not allow `continue` or `break` out of loops in the first place, the `isContinue()` does the trick. The `verify` is there mostly to check that behaviour is not broken by further passes.